### DISCUSSION TITLE
[v5] legend positioning

### DISCUSSION THREAD

**Discussion by clmz** [antvis/G2#4986] [Q&A] [answered]
how can I set the position of the legend below the chart, e.g. in v4:
```
chart.legend({
position: 'bottom-left',
})
```
but this (and many many more properties) don't work anymore and it seems that there's no doc for this?

futher:
What leads to a cropped itemLabel?

![image](https://user-images.githubusercontent.com/47241471/222755062-dc130af6-a3cd-4013-9cfb-9d238c4abfde.png)

when my code is like this:

`.legend('color', {
            itemLabel: 'Total',
            layout: 'flex',
            width: 1000,
          });`

**Comment by Aarebecca** [comment, marked as answer]
In v5, there are two separate configuration items for layout: position and layout. 
The `position` configuration determines where the components are located within different areas of the icon, such as up, down, left, and right. 
The `layout` configuration, on the other hand, determines how the components are arranged within their respective areas.

For example, in the following code snippet, we position the legend to the left, which may result in a crowded space. To solve this issue, we adjust the size of the legend to provide enough space.

```js
.legend('color', {
    size: 100,
    position: 'left'
  })
```

<img width="600" alt="image" src="https://github.com/antvis/G2/assets/25787943/17f430e8-e3fe-4d88-8936-776203d013aa">

If we want to achieve the layout shown in the bottom left of the image, we can use the "layout" configuration. Currently, v5 supports flex layouts, and we can configure a layout like this:

```js
  .legend('color', {
    size: 100,
    position: 'left',
    layout: {
      alignItems: 'flex-end'
    }
  })
```

This will align the items within the legend to the bottom of the area, as shown in the image.

<img width="827" alt="image" src="https://github.com/antvis/G2/assets/25787943/5e0a043f-2f42-452d-94e4-d09c23f51909">
